src/files/cat.js

```diff
@@ -13,7 +18,9 @@ module.exports = (send) => {
     }
 
     try {
-      hash = cleanCID(hash)
+      if (checkValidInput(hash)) {
+        hash = cleanCID(hash)
+      }
```

Comment:
I think you're right, we should delegate the parameter checking to the IPFS node we're calling so not to duplicate the logic and have to keep two sets of the same code in sync. We just need to serialize valid inputs that aren't already serialized. Could we simply do something like this?:

```js
try {
  hash = typeof hash === 'string' ? hash : cleanCID(hash)
} catch (err) {
  return callback(err)
}
```

Reply:
Sure that could work I'll update the and also look into if and how this can be tested.